### ISSUE TITLE
Choosing "Full Scans" in "Query Activity" report results in "Division by zero" error

### ISSUE THREAD

**Issue by vqmarkman** [sundeck-io/OpsCenter#187]
Traceback:

```
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 565, in _run_script
    exec(code, module.__dict__)
File "/tmp/appRoot/pages/02_Queries.py", line 19, in <module>
    reports.display(options)
File "/tmp/appRoot/./reports.py", line 30, in display
    options[report](filter_values, credit_cost)
File "/tmp/appRoot/./reports_query_activity.py", line 139, in report
    overview()
File "/tmp/appRoot/./reports_query_activity.py", line 94, in overview
    df = connection.execute_select(
File "/tmp/appRoot/./connection.py", line 142, in execute_select
    return Connection.execute(sql, args, is_select=True)
File "/tmp/appRoot/./connection.py", line 127, in execute
    return cls.get().sql(sql).to_pandas()
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/snowflake/snowpark/_internal/telemetry.py", line 139, in wrap
    result = func(*args, **kwargs)
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/snowflake/snowpark/dataframe.py", line 765, in to_pandas
    result = self._session._conn.execute(
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/snowflake/snowpark/_internal/server_connection.py", line 442, in execute
    result_set, result_meta = self.get_result_set(
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/snowflake/snowpark/_internal/analyzer/snowflake_plan.py", line 180, in wrap
    raise ne.with_traceback(tb) from None
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/snowflake/snowpark/_internal/analyzer/snowflake_plan.py", line 110, in wrap
    return func(*args, **kwargs)
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/snowflake/snowpark/_internal/server_connection.py", line 550, in get_result_set
    result = self.run_query(
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/snowflake/snowpark/_internal/server_connection.py", line 102, in wrap
    raise ex
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/snowflake/snowpark/_internal/server_connection.py", line 96, in wrap
    return func(*args, **kwargs)
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/snowflake/snowpark/_internal/server_connection.py", line 365, in run_query
    raise ex
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/snowflake/snowpark/_internal/server_connection.py", line 346, in run_query
    results_cursor = self._cursor.execute(query, params=params, **kwargs)
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/snowflake/connector/cursor.py", line 829, in execute
    Error.errorhandler_wrapper(
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/snowflake/connector/errors.py", line 232, in errorhandler_wrapper
    handed_over = Error.hand_to_other_handler(
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/snowflake/connector/errors.py", line 287, in hand_to_other_handler
    cursor.errorhandler(connection, cursor, error_class, error_value)
File "/usr/lib/python_udf/0c1c7b45c3ebd0109c3e2d9fac53af7f83336b4f784d87ddee9649a5f3399ad5/lib/python3.8/site-packages/snowflake/connector/errors.py", line 165, in default_errorhandler
    raise error_class(
  ```
  
Screenshot:
<img width="1911" alt="Screenshot 2023-08-23 at 2 43 36 PM" src="https://github.com/sundeck-io/OpsCenter/assets/104043363/2538e806-7f39-46e3-be8a-08c219efc23c">

I'm running with git sha **35a1176**

@jinfengni , my assessment about testing was not quite accurate. Predefined labels can be used in the reports and that's UI component. Can you please look at this with the highest priority as I believe it is a regression. Thank you!

**Comment by jinfengni** [comment]
Technically, it's an existing problem in the code.  If I add a user label with condition `
coalesce(partitions_scanned, 0)/coalesce(partitions_total, 1) > 0.95`, without importing the predefined_labels `FULL SCAN` , it also hit same error "division by zero".

**Comment by jinfengni** [comment]
The problem of `division by zero` is rows in `query_history` has `partitions_total = 0`.  As such, it would have `coalesce(partitions_total, 1)` = 0.

```
select coalesce(partitions_total, 1) , partitions_total from snowflake.account_usage.query_history limit 10;
+-------------------------------+------------------+
| COALESCE(PARTITIONS_TOTAL, 1) | PARTITIONS_TOTAL |
|-------------------------------+------------------|
|                             0 |                0 |
|                             0 |                0 |
|                             0 |                0 |
|                             0 |                0 |
|                             0 |                0 |
|                             0 |                0 |
|                             0 |                0 |
|                             0 |                0 |
|                             0 |                0 |
|                             0 |                0 |
+-------------------------------+------------------+
``` 

So, the problem is the evaluation of a particular condition would hit `division by zero` issue.

**Comment by jinfengni** [comment]
Looks like we should change the label condition for 'Full Scan' to `coalesce(partitions_scanned, 0) > 0.95 * coalesce(partitions_total, 1)`.   This would avoid the `division by zero` error.

**Comment by jinfengni** [comment]
In general, we have `labeld_query_history` defined using 

```
case when condition then true else false end as  label_name
```

When the expression of `condition` hit some evaluation error for one row in `query_history`,  the view evaluation would fail with such error.

**Comment by rymurr** [comment]
closed with #189
